core/templates/dev/head/pages/exploration_editor/editor_tab/StateInteraction.js

```diff
@@ -303,7 +303,16 @@ oppia.controller('StateInteraction', [
                 }
               };
 
+              $scope.$on('existingRulesChange', function(data) {
+                $scope.changedRules = data.targetScope.newAnswers;
+              });
+
               $scope.save = function() {
+                $scope.interactionId = stateInteractionIdService.savedMemento;
+                if ($scope.interactionId === 'MultipleChoiceInput') {
```

Comment:
I would try to find a more generic approach that does not require referring to specific interaction IDs (like 'MultipleChoiceInput') in the part of the code that's supposed to be treating the general case.

Reply:
Okay, I could check for ```$scope.changedRules``` is defined and replace the interaction checking part, would that be fine?
Other than that, is using an event emitter a fair approach?

Reply:
I don't think it's a good idea to have a variable that may or may not be defined, also. That sort of code is hard to maintain.

What's wrong with the original approach I suggested? That approach seemed fairly generic IMO.

Reply:
Actually I wasn't able to find the .displayed and .savedMemento properties for rules. Instead I was able to discover it for choices but not rules. If you could tell me where it's relevant to search, that would be great! Thanks!

Reply:
Hi @seanlip, I searched for customisationArgsMemento and that for AnswerGroupsMemento in ExplorationStatesService. I noticed that I still need to send the updated answerGroup to save it further. Am I looking in the right place?

Reply:
Sorry, I'm going to need some time to get back to you on this. I agree with you that the .displayed etc. does not seem to exist for the rules and I'm not sure why, but I don't have time to look into it at the moment!

Reply:
Okay! Meanwhile I'll fix the issues for what tests are failing or try to find a more generic approach for the issue.